tools/functions.sh

```diff
@@ -109,7 +109,9 @@ do_curl_tarball() {
     dldir="$(mktempdir)"
     mkdir "${dldir}/${cmd}"
     do_curl "${dldir}/${cmd}.tar.gz" "${url}"
-    if ! ${checksums_path} == ""; then 
+    ## if checksum_path is set validate tarball
+    if [ ${checksum_path:+1} ]
```

Comment:
This works but seems a bit subtle. I had to go back and relearn what `:+` does :-)  I think the most common idiom here would be:

`if [ -n "${checksum_path}" ]`